### ISSUE TITLE
Search Markets

### ISSUE THREAD

**Issue by scottzer0** [AugurProject/augur#153]
Allow a user to search the collection of markets and sort by various criteria

**Comment by thinkloop** [comment]
I've completed a solution for the "search" functionality, and separated the "sort" functionality into its own distinct issue: #206.

To realize the solution, I created a SearchStore and a SearchActions components. These can/should be leveraged for the "sort" solution as well.

I paid special attention to the following details:

* debounced input so there are no hiccups as the user types

* markets are continuously updated, so the search criteria needs to be re-applied on all changes

* pagination needs to be reset upon search so that users don't end up on non-existent pages, or in the middle of the result set

* changed body css from overflow-y:auto to overflow-y: scroll so that the scroll bar doesn't make the page jump around on different results (by disappearing and reappearing)

I am currently waiting for access to the repo to send the PR.
